js/coinsbit.js

```diff
@@ -0,0 +1,799 @@
+                        'max': undefined,
+                    },
+                    'price': {
+                        'min': undefined,
+                        'max': undefined,
+                    },
+                    'cost': {
+                        'min': this.safeNumber (market, 'feePrec'),     // <<< TODO >>> I am not yet sure this is correct.
+                        'max': undefined,
+                    },
+                },
+                'info': market,
+            });
+        }
+        return result;
+    }
+
+    async fetchTicker (symbol = undefined, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'market': market['id'],
+        };
+        let ticker = undefined;
+        const response = await this.publicGetTicker (request);
+        //     {
+        //          success: true,
+        //          message: "",
+        //          result: {
+        //            bid: '46504.7412435',
+        //            ask: '46511.54203562',
+        //            open: '47030.33',
+        //            high: '47549.99',
+        //            low: '45571.08',
+        //            last: '46459.99',
+        //            volume: '8372.29022355',
+        //            deal: '388090406.73607745',
+        //          },
+        //          code: "200"
+        //     }
+        ticker = this.safeValue (response, 'result', {});
+        return this.parseTicker (ticker, market);
+    }
+
+    async fetchTickers (symbols = undefined, params = {}) {
+        const request = params;
+        await this.loadMarkets ();
+        let tickers = {};
+        const response = await this.publicGetTickers (request);
+        //     {
+        //          success: true,
+        //          message: "",
+        //          result: {
+        //            BTC_USDT: {
+        //               at: '1640028811',
+        //               ticker: {
+        //                 bid: '46448.49682539',
+        //                 ask: '46458.41093569',
+        //                 open: '46687.39',
+        //                 high: '47550',
+        //                 low: '45584.27',
+        //                 last: '46462.77',
+        //                 vol: '32.56275846',
+        //                 deal: '1510574.62551143',
+        //                 change: '-0'
+        //               }
+        //            },
+        //            ...
+        //          },
+        //          code: "200"
+        //     }
+        tickers = this.safeValue (response, 'result', {});
+        const result = [];
+        const values = Object.values (tickers || []);
+        const keys = Object.keys (tickers);
+        for (let i = 0; i < values.length; i++) {
+            const key = keys[i];
+            const market = this.market (key);
+            result.push (this.extend (this.parseTicker (values[i], market), params));
```

Comment:
We should not extent the ticker with `params` here. And we should be using `safeMarket`.

```JavaScript
        const values = Object.values (tickers || []);
        const keys = Object.keys (tickers);
        for (let i = 0; i < values.length; i++) {
            const key = keys[i];
            const market = this.market (key);
            result.push (this.extend (this.parseTicker (values[i], market), params));
```
↓
```JavaScript
        const marketIds = Object.keys (tickers);
        for (let i = 0; i < marketIds.length; i++) {
            const marketId = marketIds[i];
            const market = this.safeMarket (marketId);
            result.push (this.parseTicker (tickers[marketId], market));
```